File: samples/DatabaseMigrations/DatabaseMigrations.AppHost/AppHost.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

var builder = DistributedApplication.CreateBuilder(args);

var sqlserver = builder.AddSqlServer("sqlserver")
    .WithLifetime(ContainerLifetime.Persistent);

var db1 = sqlserver.AddDatabase("db1");

var migrationService = builder.AddProject<Projects.DatabaseMigrations_MigrationService>("migration")
    .WithReference(db1)
    .WaitFor(db1);

builder.AddProject<Projects.DatabaseMigrations_ApiService>("api")
    .WithReference(db1)
    .WaitForCompletion(migrationService);

builder.Build().Run();

File: samples/database-migrations/ts/apphost.ts
import { ContainerLifetime, createBuilder } from './.modules/aspire.js';

const builder = await createBuilder();

const sqlserver = await builder.addSqlServer("sqlserver")
    .withDataVolume()
    .withLifetime(ContainerLifetime.Persistent);

const db1 = await sqlserver.addDatabase("db1");

const migrationService = await builder.addProject("migration", "../DatabaseMigrations.MigrationService/DatabaseMigrations.MigrationService.csproj", "https")
    .withReference(db1)
    .waitFor(db1);

await builder.addProject("api", "../DatabaseMigrations.ApiService/DatabaseMigrations.ApiService.csproj", "https")
    .withReference(db1)
    .waitForCompletion(migrationService);

await builder.build().run();
